federation/pkg/dnsprovider/providers/aws/route53/zone.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package route53
+
+import (
+	"github.com/aws/aws-sdk-go/service/route53"
+	"k8s.io/kubernetes/federation/pkg/dnsprovider"
+)
+
+type Zone struct {
+	impl  *route53.HostedZone
+	zones *Zones
+}
+
+func (zone *Zone) Name() string {
+	return *zone.impl.Name
+}
+
+func (zone *Zone) ResourceRecordSets() (dnsprovider.ResourceRecordSets, bool) {
```

Comment:
It implements the dnsprovider.Zone interface. I should make that explicit.  Documentation about what the bool is for is there (although it's mostly a hangover from the cloudprovider interface, where cloudproviders can decide to return false rather than support parts of the interface.  I'll remove that in a separate PR, as it has several knock-on effects which I'd rather postpone.